### PR TITLE
iomp5, p4c4, misc.

### PR DESCRIPTION
**Pull request by loriab**
## Description
Much misc.

## Todos
  - [x] Better error msg for MathOpenMP, mostly when GCC+MKL and can't find iomp5
  - [x] Fix items in p4c4 interface
    - [x] ccsd(t) grad in cfour v1 (came down to an unneeded `\s*`). test-suite-wise, approx works equally well with v1 or v2
    - [x] DROPMO handling
    - [x] backport a lot of the parsing changes from github.com/qcdb/qcdb to here
    - [x] modernize Molecult.init_with_xyz()
    - [ ] @JonathonMisiewicz, if you hit can't-align errors, let me know, and I'll switch it out to new mech
  - [x] I forgot to obfuscate the global variable names used in pylibefp in #953, so do that here
  - [x] fix [forum 1009](http://forum.psicode.org/t/nameerror-name-wfn-is-not-defined/1009/2?u=loriab)
  - [x] update a bunch of docs items that had accumulated
  - [x] officially make python 2.7 unacceptable to the build system

## Checklist
- [ ] ~[All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)~

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by JonathonMisiewicz** [comment]
Testing confirms this fixes all the p4c4 problems I needed fixed. Check my item off the list.

**Comment by loriab** [comment]
I think this is RTG. @JonathonMisiewicz's remaining bug report is getting "deferfix"ed. This PR has some docs stuff I'd like to get in before the Friday hackathon.

File: tests/cfour/psi-ghost-grad/input.dat [hunk 1]
```diff
@@ -67,17 +39,35 @@ set geom_maxiter 3
...
 compare_matrices(grad_p4_mA, psi4.get_gradient(), 5, 'P4 vs C4 gradient w/ghosts')  #TEST
 
 clean()
 opt_clean()
 
 activate(methdim)
+try:
+    optimize('mp2', return_wfn=True)
+except psi4.OptimizationConvergenceError as oce:
+    grad_p4_mB = oce.wfn.gradient()
```
Comment (andysim) [inline tests/cfour/psi-ghost-grad/input.dat]: I'm a little confused here; it looks like `grad_p4_mB` is only defined if the exception is raised, but is unconditionally checked below.  Should the check below be guarded to see if `grad_p4_mB` is defined?

File: CMakeLists.txt [hunk 2]
```diff
@@ -162,7 +162,7 @@ message(STATUS "Psi4 install: ${CMAKE_INSTALL_PREFIX}")
 # * library: not at all
 
 set(Python_ADDITIONAL_VERSIONS 3.7 3.6 3.5)  # adjust with CMake minimum FindPythonInterp
-find_package(PythonLibsNew 2.7 REQUIRED)
```
Comment (andysim) [inline CMakeLists.txt]: Nice!

File: tests/cfour/psi-ghost-grad/input.dat (hunk 1 above)
Comment (loriab) [reply tests/cfour/psi-ghost-grad/input.dat]: I think it's fine. The point is to get a non-near-zero gradient, so unless optking gets really good and finishes w/i 3 cycles, the exception should always be raised.

**Pull request merged**